driver-kafka/src/main/java/io/openmessaging/benchmark/driver/kafka/KafkaBenchmarkConsumer.java

```diff
@@ -18,35 +18,65 @@
  */
 package io.openmessaging.benchmark.driver.kafka;
 
-import java.util.HashMap;
-import java.util.Map;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-
+import io.openmessaging.benchmark.driver.BenchmarkConsumer;
+import io.openmessaging.benchmark.driver.ConsumerCallback;
 import org.apache.kafka.clients.consumer.ConsumerRecord;
 import org.apache.kafka.clients.consumer.ConsumerRecords;
 import org.apache.kafka.clients.consumer.KafkaConsumer;
 import org.apache.kafka.clients.consumer.OffsetAndMetadata;
 import org.apache.kafka.common.TopicPartition;
 
-import io.openmessaging.benchmark.driver.BenchmarkConsumer;
-import io.openmessaging.benchmark.driver.ConsumerCallback;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 public class KafkaBenchmarkConsumer implements BenchmarkConsumer {
 
     private final KafkaConsumer<byte[], byte[]> consumer;
+    private final String topic;
 
     private final ExecutorService executor;
+    private int partitions;
 
-    public KafkaBenchmarkConsumer(KafkaConsumer<byte[], byte[]> consumer, ConsumerCallback callback) {
+    public KafkaBenchmarkConsumer(KafkaConsumer<byte[], byte[]> consumer, String topic, int partitions) {
         this.consumer = consumer;
+        this.topic = topic;
+        this.partitions = partitions;
         this.executor = Executors.newSingleThreadExecutor();
+    }
 
+    @Override
+    public void close() throws Exception {
+        try {
+            consumer.close();
+        } finally {
+            executor.shutdownNow();
+        }
+    }
+
+    @Override
+    public CompletableFuture<Void> receiveAsync(ConsumerCallback callback) {
+        CompletableFuture<Void> future = new CompletableFuture<>();
         this.executor.execute(() -> {
             while (true) {
+
                 ConsumerRecords<byte[], byte[]> records = consumer.poll(100);
+
+                // If we ran out of records to consume, start from beginning (TODO?).
+                /*if (records.isEmpty()) {
+                    for (int partition = 0; partition < partitions; partition++) {
+                        consumer.seekToBeginning(Collections.singleton(new TopicPartition(topic, partition)));
```

Comment:
I'm thinking of adding a separate invocation of consumer after the regular, current produce-consume stops running. The separate produce-consume will seekFromBegining in order to simulate a DR event or simply how a consumer will perform in "catch-up" mode. This test would have the following metrics: messages/sec, MB/s, and ordered-completeness. 

Thoughts?